component-api/src/main/java/org/talend/sdk/component/api/record/Schema.java

```diff
@@ -63,16 +66,41 @@ default Entry getEntry(final String name) {
     String getProp(String property);
 
     enum Type {
-        RECORD,
-        ARRAY,
-        STRING,
-        BYTES,
-        INT,
-        LONG,
-        FLOAT,
-        DOUBLE,
-        BOOLEAN,
-        DATETIME
+        RECORD(new Class<?>[] { Record.class }),
+        ARRAY(new Class<?>[] { Collection.class }),
+        STRING(new Class<?>[] { String.class }),
+        BYTES(new Class<?>[] { byte[].class, Byte[].class }),
+        INT(new Class<?>[] { Integer.class }),
+        LONG(new Class<?>[] { Long.class }),
+        FLOAT(new Class<?>[] { Float.class }),
+        DOUBLE(new Class<?>[] { Double.class }),
+        BOOLEAN(new Class<?>[] { Boolean.class }),
```

Comment:
Shouldn't we include primitives classes as in MappingUtils? (ie int.class, etc.)

Reply:
As "public boolean isCompatible(final Object input)" as an Object in input, a call with a primitive will be "boxed" into corresponding object; so primitives type is not useful.